app/controllers/carts_controller.rb

```diff
@@ -42,7 +45,7 @@ def destroy
   private
 
   def grand_total
-    @cart_items.reduce(0){|sum, cart_item| sum += cart_item.total}
+    OrderItem.where(id: session[:cart]).reduce(0){|sum, cart_item| sum += cart_item.total}
```

Comment:
Useless assignment to variable - `sum`. Use just operator `+`.<br>Line is too long. [90/80]<br>Prefer `inject` over `reduce`.<br>Space missing to the left of {.<br>Space between { and | missing.<br>Space missing inside }.